src/ChatPage.cpp

```diff
@@ -189,13 +189,25 @@ ChatPage::ChatPage(QSharedPointer<UserSettings> userSettings, QWidget *parent)
         connect(this, &ChatPage::syncUI, this, [this](const mtx::responses::Rooms &rooms) {
                 view_manager_->sync(rooms);
 
-                bool hasNotifications = false;
+                static unsigned int prevNotificationCount = 0;
+                unsigned int notificationCount            = 0;
                 for (const auto &room : rooms.join) {
-                        if (room.second.unread_notifications.notification_count > 0)
-                                hasNotifications = true;
+                        notificationCount += room.second.unread_notifications.notification_count;
                 }
 
-                if (hasNotifications && userSettings_->hasNotifications())
+                // If we had less notifications last time we checked, send an alert if the user
+                // wanted one. Technically, this may cause an alert to be missed if new ones come in
+                // while you are reading old ones. Since the window is almost certainly open in this
+                // edge case, that's probably a non-issue.
```

Comment:
```suggestion
                // HACK: If we had less notifications last time we checked, send an alert if the user
                // wanted one. Technically, this may cause an alert to be missed if new ones come in
                // while you are reading old ones. Since the window is almost certainly open in this
                // edge case, that's probably a non-issue.
                // TODO: Replace this once we have proper pushrules support. This is a horrible hack
```